src/test/lib/HandshakeTest.cpp

```diff
@@ -4207,14 +4173,15 @@ QuicTestConnectionPoolCreate(
             Contexts[i].PartitionIndex = 0;
         }
 
-        ServerMultiAcceptContext ServerAcceptContext {};
-        ServerAcceptContext.MaxConnections = NumberOfConnections;
-        ServerAcceptContext.Connections = ServerConnections.get();
-        ServerAcceptContext.StartedConnectionCount = 0;
-        TestListener Listener(Registration, ServerMultiAcceptContext::ListenerMultiAcceptConnection, ServerConfiguration);
-        TEST_TRUE(Listener.IsValid());
-        Listener.Context = &ServerAcceptContext;
-        TEST_QUIC_SUCCEEDED(Listener.Start(Alpn, &ServerAddr.SockAddr));
+        MsQuicAutoAcceptListener Listener(Registration, ServerConfiguration, MsQuicConnection::NoOpCallback);
+        TEST_QUIC_SUCCEEDED(Listener.GetInitStatus());
+
+        if (TestCibirSupport) {
+            TEST_QUIC_SUCCEEDED(Listener.SetCibirId(CibirId, CibirIdLength));
+        }
+
+        TEST_QUIC_SUCCEEDED(Listener.Start(Alpn, ServerAddr));
+        TEST_QUIC_SUCCEEDED(Listener.GetLocalAddr(ServerAddr));
```

Comment:
Just curious, why do we need to get the address? Is it not fully set when we start the listener?

Reply:
`ServerAddr` doesn't have a port number set before it is passed to `Listener.Start`, this allows the networking stack to select an available port number for the listener. Since we need to tell the clients which port to connect to, we have to ask the listener the full address+port that it is listening on to get the port number.

Not setting the port number is a test reliability pattern; compile-time chosen ports occasionally fail in automation when the port is in use on a test machine. This way, the listener will always get an available port.

Reply:
Makes sense. I saw a Listener, I assumed the port had to be specified :)